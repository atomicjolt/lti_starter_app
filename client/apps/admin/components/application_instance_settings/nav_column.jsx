import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router3';
import _ from 'lodash';

export default function NavColumn(props) {

  const {
    location,
    application,
    applicationInstance,
  } = props;

  const path = `/applications/${application.id}/application_instances/${applicationInstance.id}/settings`;
  const currentPath = location.pathname;

  const tabs = [
    {
      name: 'Overview',
      iconClass: 'far fa-analytics',
      selected: currentPath === `${path}/analytics` || currentPath === path,
      route: `${path}/analytics`,
    },
    {
      name: 'General Settings',
      iconClass: 'far fa-edit',
      selected: currentPath === `${path}/generalSettings`,
      route: `${path}/generalSettings`,
    },
    {
      name: 'LTI Advantage',
      iconClass: 'far fa-edit',
      selected: currentPath === `${path}/ltiAdvantage`,
      route: `${path}/ltiAdvantage`,
    },
    {
      name: 'Configuration',
      iconClass: 'far fa-cog',
      selected: currentPath === `${path}/configuration`,
      route: `${path}/configuration`,
    },
    {
      name: 'XML Config',
      iconClass: 'far fa-cog',
      selected: currentPath === `${path}/xmlConfig`,
      route: `${path}/xmlConfig`,
    },
  ];

  return (
    <div className="nav-col">
      {_.map(tabs, (tab, index) => (
        <Link key={index} to={tab.route} className={`tabs ${tab.selected && 'active'}`}>
          <i className={`${tab.iconClass} icon`} />
          <p className="text">
            {tab.name}
          </p>
        </Link>
      ))}
    </div>
  );
}

NavColumn.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string
  }),
  application: PropTypes.shape({
    id: PropTypes.number
  }),
  applicationInstance: PropTypes.shape({
    id: PropTypes.number
  }),
};
