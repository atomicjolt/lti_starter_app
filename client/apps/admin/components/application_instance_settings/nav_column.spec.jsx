import React from 'react';
import TestRenderer from 'react-test-renderer';

import { nav_column } from './nav_column';

describe('nav_column', () => {
  it('renders the nav_column component', () => {
    const result = TestRenderer.create(
        <nav_column />
    );
    expect(result).toMatchSnapshot();
  });
});
