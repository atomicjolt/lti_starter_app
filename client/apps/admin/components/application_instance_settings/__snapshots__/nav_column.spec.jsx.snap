// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav_column renders the nav_column component 1`] = `<nav_column />`;
